cpp/src/arrow/dataset/file_csv_test.cc

```diff
@@ -50,7 +50,10 @@ class CsvFormatHelper {
   }
 
   static std::shared_ptr<CsvFileFormat> MakeFormat() {
-    return std::make_shared<CsvFileFormat>();
+    auto format = std::make_shared<CsvFileFormat>();
+    // Required for CountRows
+    format->parse_options.ignore_empty_lines = false;
```

Comment:
I'm not sure I understand this change. Is `CountRows` supposed to count logical rows of data, or physical rows inside the file (even if they may be ignored as empty)?

Reply:
It is logical rows of data; I'll clarify this.

Reply:
Hmm, but then why is this change necessary? Is it just a byproduct of how the tests are written?

Reply:
Ah, I see you added an explanation, sorry!